src/sources/nats.rs

```diff
@@ -73,6 +73,11 @@ pub struct NatsSourceConfig {
     #[serde(default = "default_decoding")]
     #[derivative(Default(value = "default_decoding()"))]
     decoding: DeserializerConfig,
+
+    /// The `NATS` subject key.
+    #[serde(default = "default_subject_key_field")]
+    subject_key_field: String,
```

Comment:
It's not clear to me from the NATS documentation: can clients specify an arbitrary field on a message to act as the subject field?

Reply:
So depending on this config option event produced by a message may look like this 
```
{
  "message": "asd",
  "nats_subject": "test.321",
  "source_type": "nats",
  "timestamp": "2022-12-04T14:32:22.696902Z"
}
```
or like this
```
{
  "message": "asd",
  "some_other_name": "test.321",
  "source_type": "nats",
  "timestamp": "2022-12-04T14:32:22.696902Z"
}```

Reply:
Gotcha, OK... so NATS clients are meant to agree on this before hand? Like it's a client-level thing where the producing client says "use field X for storing the subject key in" and any clients consuming those messages also say "grab messages with subject value "foo", which is stored in field X" and that's how they coordinate?

Given that the NATS docs have a specific section on subjects, it seems obvious that it's more than "NATS doesn't know about this field at all", even if that just means it's a common convention used by NATS clients.

Reply:
Sorry if I'm not speaking clearly enough, I'll start all over, because I feel like we're not understanding each other, which is probably mostly my fault.

So, NATS has a subject property on a message and it's not part of message at all.
Our NATS source supports reading from any subject and it's configured via [this option](https://github.com/makarchuk/vector/blob/NATS-subject-key/src/sources/nats.rs#L51).

But since NATS supports wildcards we can listen on subject like `ORDERS.*` and receive messages from both `ORDERS.DELIVERED` and `ORDERS.CANCELED`. And when vector receives such a subject we would like to be able to tell the difference between those two events. So we're adding to original subject to a message.

It happens outside of NATS and [inside of vector](https://github.com/makarchuk/vector/blob/NATS-subject-key/src/sources/nats.rs#L187-L194), NATS doesn't know about this at all.

And we give our user a chance to change a name for this key, in case it clashes with something or they just like some other name better.

That's what this config option changes.

Name of the key on Vector event that will hold NATS subject name.